### PR TITLE
incorporate rollup into build system

### PR DESCRIPTION
**Pull request by mohsinulhaq**
two things to do here:
- eliminate propTypes in production mode only
- somehow remove propTypes global from iife build

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -1,35 +1,15 @@
...
+    ['@babel/react']
   ],
-  [
-    '@babel/react',
-    {
-      useBuiltIns: true
-    }
+  plugins: [
+    '@babel/plugin-proposal-class-properties',
+    'transform-react-remove-prop-types'
```
Comment (mohsinulhaq) [inline babel.config.js]: need to do this only for prod build

Comment (denisborovikov) [reply babel.config.js]: Can use NODE_ENV for that.

Comment (denisborovikov) [reply babel.config.js]: Seems rollup doesn't use it anyway and have it's own babel config.

Comment (mohsinulhaq) [reply babel.config.js]: can u add that?

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { terser } from 'rollup-plugin-terser';
+import { sizeSnapshot } from 'rollup-plugin-size-snapshot';
+import pkg from './package.json';
+
+const input = 'src/index.js';
+const name = 'TooltipTrigger';
+const globals = {
+  react: 'React',
+  'react-dom': 'ReactDOM',
+  'prop-types': 'PropTypes',
```
Comment (mohsinulhaq) [inline rollup.config.js]: need it to not be a global when in prod mode

Comment (denisborovikov) [reply rollup.config.js]: As I understand everything here is prod mode. For dev mode, we can use docz and it has its own config.

Comment (mohsinulhaq) [reply rollup.config.js]: let me make it clear first.
for iife build, i have .js and .min.js for dev and prod mode.
and for cjs and umd, i have made no optimization and the user's module bundler will handle that (hopefully)
So what remains is removing prop-type global from min.js file.
the usage removal is taken care of by babel.

Comment (mohsinulhaq) [reply rollup.config.js]: now if u remove prop type from global object, i bundles that package into our final build. 
So i dont know what to do.

**Comment by mohsinulhaq** [comment]
https://github.com/facebook/prop-types/blob/master/README.md#how-to-depend-on-this-package
no need to remove the package

**Pull request merged**